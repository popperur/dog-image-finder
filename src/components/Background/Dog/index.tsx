import {
  BackLeg,
  BackLegFoot,
  DogBackTorso,
  DogContainer,
  FrontLeftFoot,
  FrontLeftLeg,
  FrontRightFoot,
  FrontRightLeg,
} from "components/Background/Dog/styles.tsx";
import DogHead from "components/Background/Dog/DogHead.tsx";
import DogTail from "components/Background/Dog/DogTail.tsx";

function Dog() {
  return (
    <DogContainer>
      <FrontLeftLeg>
        <FrontLeftFoot />
      </FrontLeftLeg>
      <DogBackTorso />
      <DogHead />
      <DogTail />
      <BackLeg>
        <BackLegFoot />
      </BackLeg>
      <FrontRightLeg>
        <FrontRightFoot />
      </FrontRightLeg>
    </DogContainer>
  );
}

export default Dog;
