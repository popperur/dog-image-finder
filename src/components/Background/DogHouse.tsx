import {
  DogHouseContainer,
  RoofLeft,
  RoofRight,
  RoofRightWall,
  WallLeft,
  WallRight,
  WallRightDoor,
} from "components/Background/styles.tsx";

function DogHouse() {
  return (
    <DogHouseContainer>
      <RoofRight>
        <RoofRightWall />
      </RoofRight>
      <WallLeft />
      <WallRight>
        <WallRightDoor />
      </WallRight>
      <RoofLeft />
    </DogHouseContainer>
  );
}

export default DogHouse;
