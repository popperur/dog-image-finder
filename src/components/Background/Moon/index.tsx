import {
  MoonCenter,
  MoonContainer,
} from "components/Background/Moon/styles.tsx";
import MoonCraters from "components/Background/Moon/MoonCraters";
import MoonGlows from "components/Background/Moon/MoonGlows";

function Moon() {
  return (
    <MoonContainer>
      <MoonGlows />
      <MoonCenter>
        <MoonCraters />
      </MoonCenter>
    </MoonContainer>
  );
}

export default Moon;
