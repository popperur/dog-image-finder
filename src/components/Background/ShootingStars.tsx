import {
  ShootingStar,
  ShootingStarContainer,
} from "components/Background/styles.tsx";
import useAnimationTimer from "hooks/useAnimationTimer.ts";

function ShootingStars() {
  const [keyIndex, animationDuration] = useAnimationTimer({
    initial: { duration: 4, delay: 6 },
    durationFn: () => 4 + Math.random() * 4,
    delayFn: () => 6 + Math.random() * 14,
  });

  return (
    <ShootingStarContainer>
      <ShootingStar
        $duration={animationDuration}
        key={`shooting-star-${keyIndex}`}
      />
    </ShootingStarContainer>
  );
}

export default ShootingStars;
