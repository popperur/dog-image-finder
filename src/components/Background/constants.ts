export const TOOLTIP_PAUSED =
  "Radio signal from the Moon! This incredible discovery could greatly enhance your experience in browsing for dogs!";
export const TOOLTIP_PLAYING = "Auuuuuu!";

// top, left, width, height, animationDelay
export const STARS = [
  [45, 66, 6, 6, 2],
  [70, 48, 5, 5, 4],
  [4, 44, 3, 3, 1],
  [23, 53, 1, 1, 2],
  [26, 93, 3, 3, 1.3],
  [46, 13, 4, 4, 2.3],
  [66, 34, 2, 2, 1.3],
  [36, 3, 3, 3, 3.3],
  [76, 54, 1, 1, 2.3],
  [56, 24, 3, 3, 1.3],
  [76, 25, 3, 3, 0.3],
  [56, 55, 1, 1, 2.31],
  [36, 35, 3, 3, 1.31],
  [86, 45, 2, 2, 2.31],
  [82, 85, 3, 3, 0.31],
  [81, 15, 1, 1, 1.31],
  [83, 35, 3, 3, 2.31],
  [23, 55, 2, 2, 4],
  [53, 50, 1, 1, 1.31],
  [83, 20, 1, 1, 1.51],
];

// top, left, width, height, opacity
export const MOON_CRATERS = [
  [23, 33, 17, 17, 0.5],
  [48, 10, 30, 30, 0.7],
  [90, 14, 20, 20, 0.4],
  [18, 63, 25, 25, 0.6],
  [33, 105, 10, 10, 0.8],
  [82, 55, 14, 14, 0.9],
  [122, 34, 20, 20, 0.6],
  [102, 84, 16, 16, 0.8],
  [63, 113, 20, 20, 0.9],
  [116, 119, 12, 12, 0.6],
];

// top, left, width, height, animationDuration, animationDelay
export const MOON_GLOWS = [
  [20, 30, 260, 260, 3, 0],
  [-20, -10, 340, 340, 3, 0.5],
];

// bottom, left, width, height
export const HILLS = [
  [183, 120, 20, 239],
  [259, 229, 28, 255],
  [211, 334, 28, 246],
];
