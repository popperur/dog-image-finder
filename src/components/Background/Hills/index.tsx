import {
  BackHill,
  HillContainer,
} from "components/Background/Hills/styles.tsx";
import Hill from "components/Background/Hills/Hill.tsx";
import { HILLS } from "components/Background/constants.ts";

function Hills() {
  const hillsProps = HILLS.map(([bottom, left, width, height]) => ({
    bottom,
    left,
    width,
    height,
  }));

  return (
    <HillContainer>
      {hillsProps.map((hillProps, index) => (
        <Hill key={index} {...hillProps} />
      ))}
      <BackHill />
    </HillContainer>
  );
}

export default Hills;
