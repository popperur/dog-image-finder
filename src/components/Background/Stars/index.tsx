import Star from "components/Background/Stars/Star.tsx";
import { StarContainer } from "components/Background/Stars/styles.tsx";
import { STARS } from "components/Background/constants.ts";

function Stars() {
  const starsProps = STARS.map(
    ([top, left, width, height, animationDelay]) => ({
      top,
      left,
      width,
      height,
      animationDelay,
    }),
  );

  return (
    <StarContainer>
      {starsProps.map((starProps, index) => (
        <Star key={index} {...starProps} />
      ))}
    </StarContainer>
  );
}

export default Stars;
