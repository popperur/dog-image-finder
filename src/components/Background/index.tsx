import {
  BackgroundBlock,
  BackgroundContainer,
  Ground,
} from "components/Background/styles.tsx";
import DogHouse from "components/Background/DogHouse.tsx";
import ShootingStars from "components/Background/ShootingStars.tsx";
import Dog from "components/Background/Dog";
import Moon from "components/Background/Moon";
import Stars from "components/Background/Stars";
import Hills from "components/Background/Hills";

// The animated background is fetched from
// https://codepen.io/drexqq/pen/vvryLb/
// and reworked by randomizing animations and adding styled components.
// Original version: LeeSungHee <3

interface BackgroundProps {
  showElements: boolean;
}

function Background({ showElements }: BackgroundProps) {
  return (
    <BackgroundContainer>
      {showElements && (
        <BackgroundBlock>
          <Stars />
          <ShootingStars />
          <Moon />
          <Dog />
          <Hills />
          <DogHouse />
          <Ground />
        </BackgroundBlock>
      )}
    </BackgroundContainer>
  );
}

export default Background;
