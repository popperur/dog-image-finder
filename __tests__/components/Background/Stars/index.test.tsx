import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import Stars from "components/Background/Stars";
import { STARS } from "components/Background/constants";

describe("Stars component", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<Stars />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("shows all stars", () => {
    const { container } = render(<Stars />);

    expect(container.firstChild?.childNodes).toHaveLength(STARS.length);
  });
});
