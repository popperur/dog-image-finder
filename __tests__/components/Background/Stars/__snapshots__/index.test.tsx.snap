// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Stars component > matches the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  width: 100%;
  height: 42%;
}

.c1 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 45%;
  left: 66%;
  width: 6px;
  height: 6px;
  animation-delay: 2s;
}

.c2 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 70%;
  left: 48%;
  width: 5px;
  height: 5px;
  animation-delay: 4s;
}

.c3 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 4%;
  left: 44%;
  width: 3px;
  height: 3px;
  animation-delay: 1s;
}

.c4 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 23%;
  left: 53%;
  width: 1px;
  height: 1px;
  animation-delay: 2s;
}

.c5 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 26%;
  left: 93%;
  width: 3px;
  height: 3px;
  animation-delay: 1.3s;
}

.c6 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 46%;
  left: 73%;
  width: 4px;
  height: 4px;
  animation-delay: 2.3s;
}

.c7 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 66%;
  left: 34%;
  width: 2px;
  height: 2px;
  animation-delay: 1.3s;
}

.c8 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 36%;
  left: 83%;
  width: 3px;
  height: 3px;
  animation-delay: 3.3s;
}

.c9 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 76%;
  left: 54%;
  width: 1px;
  height: 1px;
  animation-delay: 2.3s;
}

.c10 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 56%;
  left: 64%;
  width: 3px;
  height: 3px;
  animation-delay: 1.3s;
}

.c11 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 76%;
  left: 75%;
  width: 3px;
  height: 3px;
  animation-delay: 0.3s;
}

.c12 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 56%;
  left: 55%;
  width: 1px;
  height: 1px;
  animation-delay: 2.31s;
}

.c13 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 36%;
  left: 35%;
  width: 3px;
  height: 3px;
  animation-delay: 1.31s;
}

.c14 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 86%;
  left: 45%;
  width: 2px;
  height: 2px;
  animation-delay: 2.31s;
}

.c15 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 82%;
  left: 85%;
  width: 3px;
  height: 3px;
  animation-delay: 0.31s;
}

.c16 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 81%;
  left: 75%;
  width: 1px;
  height: 1px;
  animation-delay: 1.31s;
}

.c17 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 83%;
  left: 35%;
  width: 3px;
  height: 3px;
  animation-delay: 2.31s;
}

.c18 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 23%;
  left: 55%;
  width: 2px;
  height: 2px;
  animation-delay: 4s;
}

.c19 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 53%;
  left: 50%;
  width: 1px;
  height: 1px;
  animation-delay: 1.31s;
}

.c20 {
  position: relative;
  background: #f1f1f1;
  border-radius: 50%;
  animation: cQDEr 2s ease-in-out infinite;
  top: 83%;
  left: 80%;
  width: 1px;
  height: 1px;
  animation-delay: 1.51s;
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c2"
    />
    <div
      class="c3"
    />
    <div
      class="c4"
    />
    <div
      class="c5"
    />
    <div
      class="c6"
    />
    <div
      class="c7"
    />
    <div
      class="c8"
    />
    <div
      class="c9"
    />
    <div
      class="c10"
    />
    <div
      class="c11"
    />
    <div
      class="c12"
    />
    <div
      class="c13"
    />
    <div
      class="c14"
    />
    <div
      class="c15"
    />
    <div
      class="c16"
    />
    <div
      class="c17"
    />
    <div
      class="c18"
    />
    <div
      class="c19"
    />
    <div
      class="c20"
    />
  </div>
</DocumentFragment>
`;
